app/models/amazon_product.rb

```diff
@@ -1,4 +1,55 @@
+require 'time'
+require 'openssl'
+require 'base64'
+require 'nokogiri'
+require 'open-uri'
+
 class AmazonProduct < Product
 
+  ENDPOINT = "webservices.amazon.com"
+  REQUEST_URI = "/onca/xml"
+
+  # See this resource:
+  #  http://webservices.amazon.com/scratchpad/index.html#http://webservices.amazon.com/onca/xml?Service=AWSECommerceService&Operation=ItemSearch&SubscriptionId=AKIAJ64U7F3OSBNH7ERQ&AssociateTag=wishcastr-20&SearchIndex=All&Keywords=nintendo wii&ResponseGroup=Images,ItemAttributes,ItemIds,OfferSummary
+  def self.generate_amazon_uri(query)
```

Comment:
See my general comment on this PR about using [Vaccuum](https://github.com/hakanensari/vacuum) instead of hand-crafting your own bridge to the Amazon API.